nixos/modules/services/security/bitwarden_rs/default.nix

```diff
@@ -0,0 +1,177 @@
+        else key + toUpper x) "" parts);
+    in partsToKey parts;
+
+  configFile = pkgs.writeText "bitwarden_rs.env" (concatMapStrings (s: s + "\n") (
+    (concatLists (mapAttrsToList (name: value:
+      if value != null then [ "${nameToKey name}=${if isBool value then boolToString value else toString value}" ] else []
+    ) (cfg.config // {
+      inherit (cfg) dataFolder;
+      webVaultEnabled = cfg.enableWebVault;
+      webSocketEnabled = cfg.enableWebsocket;
+    })))));
+
+in {
+  options.services.bitwarden_rs = with types; {
+    enable = mkEnableOption description;
+
+    user = mkOption {
+      type = str;
+      default = "bitwarden_rs";
+      description = ''
+        User under which ${description} runs.
+      '';
+    };
+
+    group = mkOption {
+      type = str;
+      default = "bitwarden_rs";
+      description = ''
+        Group under which ${description} runs.
+      '';
+    };
+
+    dataFolder = mkOption {
+      type = str;
+      default = "/var/lib/bitwarden_rs";
+      description = ''
+        The folder under which ${description} will preserve any persistent data.
+      '';
+    };
+
+    enableWebVault = mkEnableOption "Bitwarden Web Vault" // {
+      default = true;
+    };
+
+    enableWebsocket = mkEnableOption "websocket notifications";
+
+    backupFolder = mkOption {
+      type = nullOr str;
+      default = null;
+      description = ''
+        The folder under which ${description} will backup its persistent data.
+      '';
+    };
+
+    backupUser = mkOption {
+      type = str;
+      default = "bitwarden_rs";
+      description = ''
+        User under which ${description} backups are run.
+      '';
+    };
+
+    backupGroup = mkOption {
+      type = str;
+      default = "bitwarden_rs";
+      description = ''
+        Group under which ${description} backups are run.
+      '';
+    };
+
+    backupOn = mkOption {
+      type = str;
+      default = "23:00";
+      description = ''
+        On what time ${description} backups are run.
+      '';
+    };
+
+    config = mkOption {
+      type = attrsOf (nullOr (either (either bool int) str));
+      default = {};
+      example = literalExample ''
+        {
+          signupsAllowed = true;
+          domain = https://example.com;
+          rocketPort = 8222;
+          rocketLog = "critical";
+        }
+      '';
+      description = ''
+        Config environment variables passed to the bitwarden.env file.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.bitwarden_rs.config.webVaultFolder = mkIf cfg.enableWebVault (mkDefault "${pkgs.bitwarden_rs-vault}/share/bitwarden_rs/vault");
+
+    users.extraUsers = optional (cfg.user == "bitwarden_rs") {
+      name = "bitwarden_rs";
+      inherit description;
+      group = "bitwarden_rs";
+    };
+
+    users.extraGroups = optional (cfg.group == "bitwarden_rs") {
+      name = "bitwarden_rs";
+    };
+
+    environment.systemPackages = [ pkgs.bitwarden_rs ];
```

Comment:
Since this is a server package, is there a point in including it here? Does the binary on its own serve any purpose?